crates/pallet-rewards/src/lib.rs

```diff
@@ -97,40 +212,222 @@ mod pallet {
 
     #[pallet::hooks]
     impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-        fn on_initialize(block_number: BlockNumberFor<T>) -> Weight {
-            Self::do_initialize(block_number);
-            T::WeightInfo::on_initialize()
-        }
-
         fn on_finalize(now: BlockNumberFor<T>) {
             Self::do_finalize(now);
         }
     }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Update dynamic issuance parameters
+        #[pallet::call_index(0)]
+        #[pallet::weight(T::WeightInfo::update_issuance_params(proposer_subsidy_points.len() as u32, voter_subsidy_points.len() as u32))]
+        pub fn update_issuance_params(
+            origin: OriginFor<T>,
+            proposer_subsidy_points: BoundedVec<
+                RewardPoint<BlockNumberFor<T>, BalanceOf<T>>,
+                T::MaxRewardPoints,
+            >,
+            voter_subsidy_points: BoundedVec<
+                RewardPoint<BlockNumberFor<T>, BalanceOf<T>>,
+                T::MaxRewardPoints,
+            >,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+
+            ProposerSubsidyPoints::<T>::put(proposer_subsidy_points);
+            VoterSubsidyPoints::<T>::put(voter_subsidy_points);
+
+            Ok(())
+        }
+    }
 }
 
 impl<T: Config> Pallet<T> {
-    fn do_initialize(_block_number: BlockNumberFor<T>) {
-        // Block author may equivocate, in which case they'll not be present here
-        if let Some(block_author) = T::FindBlockRewardAddress::find_block_reward_address() {
-            let reward = T::BlockReward::get();
-            let _imbalance = T::Currency::deposit_creating(&block_author, reward);
-            T::OnReward::on_reward(block_author.clone(), reward);
-
-            Self::deposit_event(Event::BlockReward {
-                block_author,
-                reward,
+    fn do_finalize(block_number: BlockNumberFor<T>) {
+        if !<T::RewardsEnabled as subspace_runtime_primitives::RewardsEnabled>::rewards_enabled() {
+            return;
+        }
+
+        if !RewardsEnabled::<T>::get() {
+            RewardsEnabled::<T>::put(true);
```

Comment:
It is stored in `pallet-subspace` in `EnableRewards` storage item, here we don't care about block number or how it is enabled exactly, just about the fact that it is. I think spec needs to be corrected @dariolina

Reply:
It is handled below already leveraging the fact that rewards are enabled once during protocol lifetime and `reward_enabled_block_number == block_number` as the result.

Reply:
I thought it stores the block number because of this part https://github.com/subspace/subspace/pull/2515/files#diff-5111fca7d87677dc2823e9608e60d1a8449f343a826171e5dc6b7d4c978939aeR1727 I misread `Some(height` as just `height`.
We need it to offset the block in reward points, but don't need to actually store it forever

Reply:
We only need to offset it before rewards are enabled. Once they are, we know the offset and set specify it from the beginning. And yes, that is what I meant when I said `pallet-subspace` stores it already.

Reply:
Okay, so if we want to update the issuance params later, the reward point pass in `update_issuance_params` is already offset with the `reward_enabled_block_number`, right?

Reply:
Not `reward_enabled_block_number` really, it will be set to the absolute block number from which the curve takes effect. These are just points defining the curve, it can start and stop anywhere.